packages/node_modules/cerebral/src/Controller.js

```diff
@@ -1,92 +1,32 @@
 import DependencyStore from './DependencyStore'
-import FunctionTree from 'function-tree'
+import BaseController from './BaseController'
 import Module from './Module'
 import Model from './Model'
 import {
   ensurePath,
-  isDeveloping,
-  throwError,
   isSerializable,
   forceSerializable,
   isObject,
-  getProviders,
   cleanPath,
+  getModule,
 } from './utils'
-import VerifyPropsProvider from './providers/VerifyProps'
-import StateProvider from './providers/State'
-import DebuggerProvider from './providers/Debugger'
-import ControllerProvider from './providers/Controller'
-import ResolveProvider from './providers/Resolve'
 
-function getModule(path, modules) {
-  const pathArray = Array.isArray(path) ? path : path.split('.')
-  return pathArray.reduce((currentModule, key) => {
-    if (!currentModule.modules[key]) {
-      throwError(
-        `The path "${pathArray.join(
-          '.'
-        )}" is invalid, can not find module. Does the path "${pathArray
-          .splice(0, path.length - 1)
-          .join('.')}" exist?`
-      )
-    }
-    return currentModule.modules[key]
-  }, modules)
-}
 /*
   The controller is where everything is attached. The devtools
   and router is attached directly. Also a top level module is created.
   The controller creates the function tree that will run all signals,
   based on top level providers and providers defined in modules
 */
-class Controller extends FunctionTree {
+class Controller extends BaseController {
   constructor(config) {
-    super()
-    const {
-      state = {},
-      signals = {},
-      providers = [],
-      modules = {},
-      router,
-      devtools = null,
-      options = {},
-    } = config
-    const getSignal = this.getSignal
-
-    this.getSignal = () => {
-      throwError(
-        'You are grabbing a signal before controller has initialized, please wait for "initialized" event'
-      )
-    }
-
-    if (router) {
-      console.warn(`DEPRECATION: the 'router' module should be in modules.`)
-      modules.router = router
-    }
+    super(
+      Object.assign(config, {
```

Comment:
maybe in reverse order? so config would be able to override default and not confuse the user.